sdk/core/azure-core/src/test/java/com/azure/core/credential/TokenCacheTests.java

```diff
@@ -28,28 +28,26 @@ public void testOnlyOneThreadRefreshesToken() throws Exception {
         CountDownLatch latch = new CountDownLatch(1);
         AtomicLong maxMillis = new AtomicLong(0);
 
-        Flux.range(1, 10)
-            .flatMap(i -> Mono.just(OffsetDateTime.now())
-                // Runs cache.getToken() on 10 different threads
-                .publishOn(Schedulers.parallel())
-                .flatMap(start -> cache.getToken()
-                    .map(t -> Duration.between(start, OffsetDateTime.now()).toMillis())
-                    .doOnNext(millis -> {
-                        if (millis > maxMillis.get()) {
-                            maxMillis.set(millis);
-                        }
-//                        System.out.format("Thread: %s\tDuration: %smillis%n",
-//                            Thread.currentThread().getName(), Duration.between(start, OffsetDateTime.now()).toMillis());
-                    })))
+        Flux.range(1, 10).flatMap(i -> Mono.fromCallable(OffsetDateTime::now))
```

Comment:
This is interesting, I thought `Mono.just()` inside a `flatMap` would require the outer publisher to be subscribed first for this `Mono.just()` to be executed.

Reply:
Looks like you're correct, this fix was then transitioning over to `parallel`. I'll update the description as it is misleading.